### PR TITLE
Policy CRD check before extracting manifests

### PR DESCRIPTION
**Pull request by jc-rh**

### CODE REVIEW

File: internal/extramanifest/extramanifest.go [hunk 1]
```diff
@@ -130,27 +132,37 @@ func (h *EMHandler) ExportExtraManifestToDir(ctx context.Context, extraManifestC
 
 // ExtractAndExportManifestFromPoliciesToDir extracts CR specs from policies. It matches policies and/or CRs by labels.
 func (h *EMHandler) ExtractAndExportManifestFromPoliciesToDir(ctx context.Context, policyLabels, objectLabels map[string]string, toDir string) error {
+
+	_, err := h.CrdClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, "policies.policy.open-cluster-management.io", metav1.GetOptions{})
+
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			h.Log.Info("Skipping extraction from policies as the policy CRD is not found")
```
Comment (omertuc) [inline internal/extramanifest/extramanifest.go:140]:
```suggestion
			h.Log.Info("Skipping extraction from policies as the policy CRD is not found <something something this is normal on non-ACM clusters>")
```

**Review by omertuc — CHANGES_REQUESTED** [review]

File: internal/extramanifest/extramanifest.go (hunk 1 above)
Comment (browsell) [reply internal/extramanifest/extramanifest.go:140]: , cluster is not ACM managed.

[the remark this answers is not in the document]
Comment (jc-rh) [reply internal/extramanifest/extramanifest.go]: Yep, already added for the test to work :)

File: internal/extramanifest/extramanifest.go [hunk 2]
```diff
@@ -144,12 +159,12 @@ func (h *EMHandler) ExtractAndExportManifestFromPoliciesToDir(ctx context.Contex
...
 		objects, err := getConfigurationObjects(policy, objectLabels)
 		if err != nil {
-			return fmt.Errorf("failed to get configuration objects: %w", err)
+			return fmt.Errorf("failed to extract manifests from policies: %w", err)
 		}
 		for _, object := range objects {
 			manifestFilePath := filepath.Join(manifestsDir, fmt.Sprintf("%d_%s_%s.yaml", i, object.GetName(), object.GetNamespace()))
-			if err := utils.MarshalToYamlFile(&object, manifestFilePath); err != nil {
-				return fmt.Errorf("failed to marshal object to file %s: %w", manifestFilePath, err)
+      if err := utils.MarshalToYamlFile(&object, filePath); err != nil {
```
Comment (browsell) [inline internal/extramanifest/extramanifest.go:166]: indention

File: internal/extramanifest/extramanifest.go [hunk 3]
```diff
@@ -130,6 +132,19 @@ func (h *EMHandler) ExportExtraManifestToDir(ctx context.Context, extraManifestC
 
 // ExtractAndExportManifestFromPoliciesToDir extracts CR specs from policies. It matches policies and/or CRs by labels.
 func (h *EMHandler) ExtractAndExportManifestFromPoliciesToDir(ctx context.Context, policyLabels, objectLabels map[string]string, toDir string) error {
+	// TODO fake the crd client instead of skipping it for UT
+	if h.CrdClient != nil {
+		_, err := h.CrdClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, "policies.policy.open-cluster-management.io", metav1.GetOptions{})
```
Comment (Missxiaoguo) [inline internal/extramanifest/extramanifest.go:137]: Quesiton, why cannot use the existing client to query CRD? The RBAC has the role

Comment (jc-rh) [reply internal/extramanifest/extramanifest.go]: Followed an old example ... Updated

**Review by donpenney — APPROVED** [review]

**Pull request merged**